bitbots_path_planning/bitbots_path_planning/planner.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+
+import numpy as np
+import pyastar2d
+import tf2_geometry_msgs   # Is there for tf to recognize geomety_msgs even if it is not used explicitly
+import tf2_ros as tf2
+from bitbots_path_planning.map import Map
+from geometry_msgs.msg import PoseStamped
+from nav_msgs.msg import Path
+from rclpy.node import Node
+from rclpy.time import Time
+from std_msgs.msg import Header
+
+
+class Planner:
+    def __init__(self, node: Node, buffer: tf2.Buffer, map: Map) -> None:
+        self.node = node
+        self.buffer = buffer
+        self.map = map
+        self.goal = None
+        self.path = None
+
+    def set_goal(self, pose: PoseStamped) -> None:
+        pose.header.stamp = Time().to_msg()
+        self.goal = pose
```

Comment:
Please swap these two lines. This is very against my intuition.

Reply:
Why? I don't understand what you mean.

Reply:
Got it. It's better this way, as there is no small moment, where there is a wrong timestamp in the class variable.